src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content-datasource/main.js

```diff
@@ -78,7 +78,7 @@ define([], function() {
                     selected: this.selected,
                     actionCallback: function(item) {
                         this.selected = item.id;
-                        this.options.selectCallback(item.id, item.path, item.title);
+                        this.options.selectCallback(item.id, item.path, item.title, item);
```

Comment:
this compoenent was introduced for the smart-content. there yo have to pass the correct values for id, fullQualifiedName and title. otherwise you have to know which value you have to use in the smart-content